### DISCUSSION TITLE
Total Stress due to torsional moment

### DISCUSSION THREAD

**Discussion by EngDM** [robbievanleeuwen/section-properties#234] [General]
Hey guys,

I've been playing around with the different stress plots WRT the Mzz moment, but I can't seem to find an accurate way to plot the total stress from the load. I have an expected value generated from the RISA 3D demo but the value I get when running section properties isn't the same or close. I'm trying to decipher the shear stress caused by torsional moment as that appears to be how most guides approach finding torsional resistance. Does anyone have more experience in what I'm trying to accomplish?

I.e. think outriggers welded to an HSS girt generating a torsional moment, increasing from zero at the middle to max at the ends.

Edit: Also, how come when using an HSS and finding it's warping constant it is so high, but HSS is usually taken with a warping constant of zero?

**Comment by robbievanleeuwen** [comment]
Hi @EngDM, 

Would you be able to post an example of your code? It would probably be helpful to post the expected value from RISA as well for comparison :)

**Regarding the warping constant:**
Although the warping constant isn't zero and its value may *seem* high, it is in fact insignificantly low in relation to how it relates to a stiffness and the torsional constant. Have a look at [this](https://gist.github.com/robbievanleeuwen/a90b2dd2c758048f82f6f8b0fc028b92) example which compares the contribution of torsional stiffness (*J*) and warping stiffness (*I_w*) to lateral torsional buckling resistance. The reference moment equation is taken from AS 4100:1998:

<img width="653" alt="Screen Shot 2022-07-15 at 2 37 36 pm" src="https://user-images.githubusercontent.com/18841910/179151657-432a4b58-91f8-4f89-a5b2-df50e56cb7e4.png">

The *GJ* term is the contribution of torsional stiffness and the following term is the contribution of warping stiffness.

The [example](https://gist.github.com/robbievanleeuwen/a90b2dd2c758048f82f6f8b0fc028b92) shows that for a 6m long I-beam, warping stiffness contributes about **24%** of the relative resistance to lateral torsional buckling. For a 6m long box section, warping contributes less than **0.004%** of the relative resistance. This is why most codes say to take the warping constant as zero for closed sections, because it might as well be!

Hopefully these numbers put the value of the warping constant into perspective!

**Reply by EngDM** [reply]
Hey thanks again.

```
#create HSS203x203x13
geometry = rectangular_hollow_section(d= 203.2, b=203.2, t=12.7, r_out=15, n_r=7)
geometry.create_mesh(mesh_sizes=[2.0])

#Create the mesh
sec = Section(geometry)
sec.plot_mesh(materials=False)
sec.calculate_geometric_properties()
sec.calculate_warping_properties()

sec.display_results(fmt='.3f')

stress_post = sec.calculate_stress(Mzz=3.203e6) #Apply the 3.203kN-m twisting
stress_post.plot_stress_mzz_zx() #finds the stress about the X-axis?
stress_post.plot_stress_zxy() #I think this is total stress

#Get the stress results

numerical_result = sum(stress_post.get_stress()[0]["sig_zxy"]) #There is a concentrated stress at the corners which throws off using just the max function.
count = len((stress_post.get_stress()[0]["sig_zxy"])) #Get around it by using the average (only works for twisting of closed symmetric sections since it is uniform)
numerical_result/count #Average calc

# test=max(stress_post.get_stress()[0]["sig_vm"]) #trying out the VM plot to see if it is any different/will give me a useable value.
# test
```

The RISA model returns 3.474MPa for the stress, and then also calculates the shear force caused in the flanges due to this stress.
![image](https://user-images.githubusercontent.com/108320854/179269030-98376d49-56f8-4912-893f-5d03a8d39997.png)

I think I might have found a way around it only for HSS and only for torsion since the mesh puts concentrated stress at the interior corners; I couldn't find a way to find the stress at the extreme fiber ignoring these peak locations. Do you know if there is a way to make the graph interactive I.E. when you hover over it, it tells you the value there? Might be a matplotlib thing I'm not sure.

**Reply by robbievanleeuwen** [reply]
The section you've analysed looks like it has an incorrect outside radius, usually the external radius for a hollow section is two times the wall thickness. These sections are usually cold formed so the internal radius you've modelled looks impractical. 

<img width="300" alt="Screen Shot 2022-07-18 at 10 26 31 am" src="https://user-images.githubusercontent.com/18841910/179430947-e2091177-a4fe-49f6-98df-b66dfa97288f.png">

Here's what an equivalent section looks like with ``r_out=25.4``:

<img width="300" alt="Screen Shot 2022-07-18 at 10 26 31 am" src="https://user-images.githubusercontent.com/18841910/179430961-dafe8d50-e728-4b9d-913e-9708b0a0b16e.png">

Having said this I don't see anything wrong with the analysis. The sharp corners **ARE** going to create a stress concentration as the torsional stress needs to change direction over a very small distance, this stress gradient causes a stress concentration and *sectionproperties* correctly models this in the results.

Stress with ``r_out=15``:

<img width="600" alt="Screen Shot 2022-07-18 at 10 29 27 am" src="https://user-images.githubusercontent.com/18841910/179431054-f2ac8f8f-813c-423b-a648-a0e77cc5b542.png">

Stress with ``r_out=25.4``:

<img width="600" alt="Screen Shot 2022-07-18 at 10 29 34 am" src="https://user-images.githubusercontent.com/18841910/179431069-4a3a40a7-8210-4cb3-a368-8e180d284f9a.png">

The screenshot from RISA tells me nothing about what you've analysed so I can't really comment on how this compares to *sectionproperties*. The average stress result returns 3.474 MPa for me which matches your RISA result, so perhaps RISA is just calculating an average stress? This is over simplifying the behaviour as you can see from the stress plots above.

At the moment there is no way to get stresses while hovering over the plot, but this may be something that could be implemented in the future once this [PR](https://github.com/robbievanleeuwen/section-properties/pull/230) is finished. You can pull down the branch if you want to have a play with the results. Further discussion and stress plots of a closed section also [here](https://github.com/robbievanleeuwen/section-properties/issues/175).
